src/mbed_tools/devices/_internal/file_parser.py

```diff
@@ -87,7 +88,42 @@ class OnlineId(NamedTuple):
     slug: str
 
 
-def read_product_code(file_contents: str) -> Optional[str]:
+@dataclass
+class DeviceFileInfo:
+    """Information gathered from Mbed device files."""
+
+    product_code: Optional[str]
+    online_id: Optional[OnlineId]
+
+
+def read_device_files(directory_paths: Iterable[pathlib.Path]) -> DeviceFileInfo:
+    """Read data from files contained on an mbed enabled device's USB MSD.
```

Comment:
I'd recommend using "mass storage device" instead of "MSD" (here and other places in this file), as MSD isn't a commonly understood acronym.